mrmustard/lab_dev/circuit_components_utils/b_to_q.py

```diff
@@ -54,3 +54,36 @@ def __init__(
             name="BtoQ",
         )
         self._add_parameter(Constant(phi, "phi"))
+        self.phi = phi
+
+    @property
+    def adjoint(self) -> BtoQ:
+        bras = self.wires.bra.indices
+        kets = self.wires.ket.indices
+        rep = self.representation.reorder(kets + bras).conj()
+
+        ret = BtoQ(self.modes, self.phi)
+        ret._representation = rep
+        ret._wires = self.wires.adjoint
+        ret._name = self.name + "_adj"
+        return ret
+
+    @property
+    def dual(self) -> BtoQ:
+        ok = self.wires.ket.output.indices
+        ik = self.wires.ket.input.indices
+        rep = self.representation.reorder(ik + ok).conj()
```

Comment:
Let's replace this by:
`ok = self.wires.ket.output.indices
ik = self.wires.ket.input.indices
ib = self.wires.bra.input.indices
ob = self.wires.bra.output.indices
rep = self.representation.reorder(ib + ob + ik + ok).conj()

ret = BtoQ(self.modes, float(self.phi.value))`